provisioner/aws-manager.js

```diff
@@ -15,9 +17,12 @@ function AwsManager(ec2, keyPrefix, pubKey) {
   this.ec2 = ec2;
   this.keyPrefix = keyPrefix;
   this.pubKey = pubKey;
-  this.__apiState = {};
-  this.__internalState = {};
   this.__knownKeyPairs = [];
+  this.__apiState = {
+    instances: [],
+    requests: [],
+  };
+  this.__internalState = [];
```

Comment:
I believe I said this in a bug comment too... But let's call it `__submittedSpotReq` or something like that, I'm sure you can give it a better name than me.

My thinking is that it's a spot request that is _submitted_ by not yet visible, so saying it's in the state _submitted_ makes some sense.